numpy/__init__.cython-30.pxd

```diff
@@ -559,11 +559,11 @@ cdef extern from "numpy/arrayobject.h":
     char * PyArray_Zero (ndarray)
     char * PyArray_One (ndarray)
     #object PyArray_CastToType (ndarray, dtype, int)
-    int PyArray_CastTo (ndarray, ndarray)
-    int PyArray_CastAnyTo (ndarray, ndarray)
-    int PyArray_CanCastSafely (int, int)
-    npy_bool PyArray_CanCastTo (dtype, dtype)
-    int PyArray_ObjectType (object, int)
+    int PyArray_CastTo (ndarray, ndarray) except -1
+    int PyArray_CastAnyTo (ndarray, ndarray) except -1
+    int PyArray_CanCastSafely (int, int)  # writes errors
+    npy_bool PyArray_CanCastTo (dtype, dtype)  # writes errors
```

Comment:
I think all these "writes errors" should either be `except *` (if you can't say anything about the return value), or `except? 0` (if errors always produce zero, but not all zeros are errors).

Reply:
Oh, "writes errors" here means it uses `PyErr_WriteUnraiseable`.  So I think for cython that should be as good as nothing.

Reply:
Oh, I thought this meant "sets the exception but doesn't indicate it in the return value".

Reply:
PyErr_WriteUnraiseable -> PyErr_WriteUnraisable

Just to help searching for the documentation :)